webui/react/src/services/decoder.ts

```diff
@@ -181,6 +183,14 @@ const ioTrialToTrial = (io: ioTypeTrialSummary): TrialSummary => {
   };
 };
 
+export const jsonToTrialDetails = (data: unknown): TrialDetails => {
+  const io = decode<ioTypeTrialDetails>(ioTrialDetails, data);
+  return { ...io,
```

Comment:
**suggestion:** I saw this pattern in the experiment detail endpoint and ran into duplicate fields wherever field names changed to camelcase due to the spread operator. For example, under experiment config, you would find both `checkpoint_storage` and `checkpointStorage` defined. It's a bit more labor but probably better to explicitly set each field without the spread operator.

Reply:
I agree I think at some point we need to write a dynamic converter that does this. we need the reverse for going back from a parsed experiment config to one that the server understands. for example when forking experiments we need to send back `checkpoint_storage` and not the camelcase version.